webclient/src/components/Message/Message.tsx

```diff
@@ -0,0 +1,103 @@
+// eslint-disable-next-line
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { NavLink, generatePath } from "react-router-dom";
+
+import { RouteEnum } from 'types';
+
+import CardCallout from './CardCallout';
+import './Message.css';
+
+// eslint-disable-next-line
+const urlRegex = /(https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b(?:[-a-zA-Z0-9@:%_\+.~#?&//=]*))/g;
+const nameRegex = /(^[^: ]+):/;
+const mentionRegex = /(@[^ ]+)/g;
+const cardCalloutsRegex = /(\[\[[^\]]+\]\])/g;
+const calloutsRegex = /(\[\[|\]\])/g;
+
+const Message = ({ message: { message, messageType, timeOf, timeReceived } }) => (
+  <div className='message'>
+    <div className='message__detail'>
+      <ParsedMessage message={message} />
+    </div>
+  </div>
+);
+
+const ParsedMessage = ({ message }) => {
+  const [messageChunks, setMessageChunks] = useState(null);
+  const [name, setName] = useState(null);
+
+  useMemo(() => {
```

Comment:
As expensive as this component may be, at least with useMemo, parsing only ever happens once per component. I don't foresee a message ever being updated within the the lifecycle of a Message component.

Reply:
I'd like better variable names / comments as these are some complicated regular expressions